great_expectations/checkpoint/util.py

```diff
@@ -239,76 +272,247 @@ def get_runtime_batch_request(
+            evaluation_parameters,
+            runtime_kwargs["evaluation_parameters"],
+        )
+        dest_config["evaluation_parameters"] = updated_evaluation_parameters
+    if runtime_kwargs.get("runtime_configuration") is not None:
+        runtime_configuration = dest_config.get("runtime_configuration") or {}
+        updated_runtime_configuration = nested_update(
+            runtime_configuration,
+            runtime_kwargs["runtime_configuration"],
+        )
+        dest_config["runtime_configuration"] = updated_runtime_configuration
+    if runtime_kwargs.get("validations") is not None:
+        validations = dest_config.get("validations") or []
+        existing_validations = source_config.get("validations") or []
+        validations.extend(
+            filter(
+                lambda v: v not in existing_validations,
+                runtime_kwargs["validations"],
+            )
+        )
+        dest_config["validations"] = validations
+    if runtime_kwargs.get("profilers") is not None:
+        profilers = dest_config.get("profilers") or []
+        existing_profilers = source_config.get("profilers") or []
+        profilers.extend(
+            filter(lambda v: v not in existing_profilers, runtime_kwargs["profilers"])
+        )
+        dest_config["profilers"] = profilers
+
+    return dest_config
+
+
+def get_updated_action_list(
+    base_action_list: list, other_action_list: list
+) -> List[dict]:
+    if base_action_list is None:
+        base_action_list = []
+
+    base_action_list_dict = {action["name"]: action for action in base_action_list}
+
+    for other_action in other_action_list:
+        other_action_name = other_action["name"]
+        if other_action_name in base_action_list_dict:
+            if other_action["action"]:
+                nested_update(
+                    base_action_list_dict[other_action_name],
+                    other_action,
+                    dedup=True,
+                )
+            else:
+                base_action_list_dict.pop(other_action_name)
+        else:
+            base_action_list_dict[other_action_name] = other_action
+
+    for other_action in other_action_list:
+        other_action_name = other_action["name"]
+        if other_action_name in base_action_list_dict:
+            if not other_action["action"]:
+                base_action_list_dict.pop(other_action_name)
+
+    return list(base_action_list_dict.values())
+
+
+def batch_request_contains_batch_data(
+    batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None
+) -> bool:
+    return (
+        batch_request is not None
+        and batch_request.get("runtime_parameters") is not None
+        and batch_request["runtime_parameters"].get("batch_data") is not None
+    )
+
+
+def batch_request_in_validations_contains_batch_data(
+    validations: Optional[List[dict]] = None,
+) -> bool:
+    if validations is not None:
+        for idx, val in enumerate(validations):
```

Comment:
It looks like enumerate is not needed here. Also, this could be refactored to call `batch_request_contains_batch_data` above.

Reply:
@NathanFarmer I left it just in case we will want the `idx` later.  If you feel we should make these changes, could you please put this idea into a separate JIRA ticket for us to follow up on?  Thanks!